stream-chat-android-docs/src/main/kotlin/io/getstream/chat/docs/kotlin/client/docusaurus/ChatDomainMigration.kt

```diff
@@ -0,0 +1,174 @@
+            if (result.isSuccess) {
+                // Handle success
+            } else {
+                // Handle error
+            }
+        }
+    }
+
+    inner class State {
+        val chatClient = ChatClient.instance()
+
+        val filter = Filters.eq("type", "messaging")
+        val sort = QuerySortByField.descByName<Channel>("lastUpdated")
+        private val queryChannelsRequest = QueryChannelsRequest(filter = Filters.eq("type", "messaging"), limit = 30)
+
+        fun accessStates() {
+            // Returns QueryChannelsState object based on filter and sort used to query channels
+            val queryChannelsState = chatClient.state.queryChannels(filter = filter, sort = sort)
+
+            // Returns ChannelState object for a given channel
+            val channelState = chatClient.state.channel(channelType = "messaging", channelId = "sampleId")
+
+            // Returns ThreadState object for a thread associated with a given parentMessageId
+            val threadState = chatClient.state.thread(messageId = "parentMessageId")
+
+            // Gives you access to GlobalState object
+            val globalState = chatClient.globalState
+        }
+
+        fun requestChannelsAndObserveTheState() {
+            // 1. Get the first 30 channels to which thierry belongs
+            val filter = Filters.`in`("members", "thierry")
+            val sort = QuerySortByField.descByName<Channel>("lastUpdated")
+            val request = QueryChannelsRequest(
+                filter = filter,
+                querySort = sort,
+                limit = 30,
+                offset = 0,
+                messageLimit = 1,
+                memberLimit = 30,
+            )
+            ChatClient.instance().queryChannels(request).enqueue { result ->
+                if (result.isSuccess) {
+                    // Request successful. Data will be propagated to the state object
+                } else {
+                    // Handle error
+                }
+            }
+            // 2. Get the state object associated with the above API call
+            val queryChannelsState = ChatClient.instance().state.queryChannels(filter = filter, sort = sort)
+        }
+
+        fun watchChannel() {
+            // Old approach - returns ChannelController object and performs watchChannel request
+            // ChatDomain.instance().watchChannel(cid = "messaging:sampleId", messageLimit = 30).enqueue { result ->
+            //     if (result.isSuccess) {
+            //         val channelController = result.data()
+            //     } else {
+            //         // Handle error
+            //     }
+            // }
+
+            // New approach - returns the StateFlow<ChannelState?> object and performs watchChannel request
+            val channelState: StateFlow<ChannelState?> = chatClient.watchChannelAsState(cid = "messaging:sampleId", messageLimit = 30, coroutineScope = scope)
```

Comment:
Maybe in the future we could format function calls with more than 2 parameters like so:

```kotlin
  val channelState: StateFlow<ChannelState?> = chatClient.watchChannelAsState(
      cid = "messaging:sampleId",
      messageLimit = 30,
      coroutineScope = scope
  )
```

It's not a big deal but I think it slightly improves readability.